airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.kt

```diff
@@ -364,9 +375,9 @@ abstract class JdbcDestinationHandler<DestinationState>(
     ): Boolean {
         // Check that the columns match, with special handling for the metadata columns.
         if (
-            !isAirbyteRawIdColumnMatch(existingTable) ||
-                !isAirbyteExtractedAtColumnMatch(existingTable) ||
-                !isAirbyteMetaColumnMatch(existingTable)
+            !(existingTable.columns.containsKey(JavaBaseConstants.COLUMN_NAME_AB_RAW_ID) && isAirbyteRawIdColumnMatch(existingTable)) ||
```

Comment:
the opposite actually - we always want to do the `existingTable.columns.containsKey`, even if a subclass overrides `isAirbyteRawIdColumnMatch`. Otherwise we'd need to copy the containsKey check into every subclass implementation.